documentation/specs/src/interoperability/ethereum-bridge/ethereum_events_attestation.md

```diff
@@ -18,15 +21,26 @@ pub enum EthereumEvent {
 }
 ```
 
-Each event will be stored with a list of the validators that have ever seen it
-as well as the fraction of total voting power that has ever seen it.
-Once an event has been seen by 2/3 of voting power, it is locked into a
+Each event will be stored with a list of the consensus validators that have
+ever seen it as well as the fraction of total voting power that has ever seen it.
+Once an event has been seen by at least 2/3 of voting power, it is locked into a
 `seen` state, and acted upon.
 
-There is no adjustment across epoch boundaries - e.g. if an event is seen by 1/3
-of voting power in epoch n, then seen by a different 1/3 of voting power in
-epoch m>n, the event will be considered `seen` in total. Validators may never
-vote more than once for a given event.
+If the voting power of Namada changes across epoch boundaries, then events in
+storage which are yet to be achieve a quorum decision behind them (i.e. whose
+`seen` state is still `false`) must have their voting power adjusted. It is
```

Comment:
How is this done, exactly? Do you mean that we don't recalculate the voting power until the event is voted again by a new validator, at which point we iterate?

Reply:
So, this is related with #1288

> Do you mean that we don't recalculate the voting power until the event is voted again by a new validator

I'm still not 100% sure how this will be implemented, however that would be the idea, yes. We store a map of a validator's adress to the block height they voted on some event, so we can scale voting power based on the earliest height we find (maybe?). Basically we need a least common multiple of some sort to then add all of the voting powers together, I think.

Reply:
fwiw I think recalculating by just re-iterating over all of the current voting powers is fine, the number of validators is a low constant.